### ISSUE TITLE
`EvaluateChatInput` should wait until chat task is complete or aborted before returning

### ISSUE THREAD

**Issue by rhennigan** [WolframResearch/Chatbook#221]
Currently, any type of chat request will return as soon as the asynchronous task is created. However, this is only necessary for `AIAutoAssist` chats, since we don't want those to block inputs.

Since `AIAutoAssist` cancels chats in progress, this results in normal in progress chat results getting cleared upon sending a new assist chat:

![AIAutoAssistCancelChat](https://github.com/WolframResearch/Chatbook/assets/6674723/d96e6587-7549-4e89-8ef9-19629c6e688a)

`EvaluateChatInput` should instead wait for its own task to complete before returning, rather than waiting for the previous task to complete before starting.
